airbyte-server/src/main/java/io/airbyte/server/handlers/MigrationHandler.java

```diff
@@ -0,0 +1,181 @@
+import io.airbyte.server.converters.ConfigConverter;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MigrationHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MigrationHandler.class);
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_data";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final String version;
+  private final ConfigRepository configRepository;
+
+  public MigrationHandler(final String version, final ConfigRepository configRepository) {
+    this.version = version;
+    this.configRepository = configRepository;
+  }
+
+  public Path exportData() {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      final Path archive = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");;
+      archive.toFile().deleteOnExit();
+      try {
+        exportVersionFile(tempFolder);
+        exportAirbyteConfig(tempFolder);
+        exportAirbyteDatabase(tempFolder);
+        ArchiveHelper.createArchive(tempFolder, archive);
+      } finally {
+        FileUtils.deleteDirectory(tempFolder.toFile());
+      }
+      return archive;
+    } catch (IOException e) {
+      LOGGER.error("Export Data failed.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportVersionFile(Path tempFolder) throws IOException {
+    final File versionFile = Files.createFile(tempFolder.resolve(VERSION_FILE_NAME)).toFile();
+    FileUtils.writeStringToFile(versionFile, version, Charset.defaultCharset());
+  }
+
+  private void exportAirbyteConfig(Path tempFolder) {
+    LOGGER.info("Exporting Airbyte Configs");
+    final ConfigConverter configConverter = new ConfigConverter(version, tempFolder);
+    Exceptions.toRuntime(() -> {
+      configConverter.writeConfigList(ConfigSchema.STANDARD_WORKSPACE,
+          List.of(configRepository.getStandardWorkspace(PersistenceConstants.DEFAULT_WORKSPACE_ID)));
+      configConverter.writeConfigList(ConfigSchema.STANDARD_SOURCE_DEFINITION, configRepository.listStandardSources());
+      configConverter.writeConfigList(ConfigSchema.STANDARD_DESTINATION_DEFINITION, configRepository.listStandardDestinationDefinitions());
+      configConverter.writeConfigList(ConfigSchema.SOURCE_CONNECTION, configRepository.listSourceConnection());
+      configConverter.writeConfigList(ConfigSchema.DESTINATION_CONNECTION, configRepository.listDestinationConnection());
+      final List<StandardSync> standardSyncs = configRepository.listStandardSyncs();
+      configConverter.writeConfigList(ConfigSchema.STANDARD_SYNC, standardSyncs);
```

Comment:
why don't you just add a `listStandrdSyncSchedules` method to the configRepository?